### PR TITLE
Support for Card Emulation

### PR DESCRIPTION
**Pull request by monich**
This requires the integration of the pending [nfcd](https://github.com/sailfishos/nfcd/pull/17) changes which in turn requires the integration of [libnfcdef](https://github.com/monich/libnfcdef) into Sailfish OS.

### CODE REVIEW

**Review by mlehtima — APPROVED** [review]

File: src/nci_adapter.c [hunk 1]
```diff
@@ -692,24 +888,16 @@ nci_adapter_create_peer_initiator(
...
             case NCI_MODE_PASSIVE_POLL_A:
                 /* NFC-DEP (Poll side) */
-                if (ntf->activation_param) {
-                    NfcParamPollA poll_a;
-
-                    return nfc_adapter_add_peer_initiator_a(adapter, target,
-                        nci_adapter_convert_poll_a(&poll_a, mp),
-                        nci_adapter_convert_nfc_dep_poll(&nfc_dep, ap));
-                }
+                peer =  nfc_adapter_add_peer_initiator_a(NFC_ADAPTER(self),
```
Comment (mlehtima) [inline src/nci_adapter.c]: This and couple of other places have extra space after =.

Comment (monich) [reply src/nci_adapter.c]: Fixed all of them))

**Comment by monich** [comment]
FYI, I found a few corner cases which need to be fixed on [libncicore](https://github.com/mer-hybris/libncicore) side before card emulation PRs can be merged.

**Comment by Thaodan** [comment]
Great work. I'm interested in this topic too. The only thing missing is either a [PC/SC](https://en.wikipedia.org/wiki/PC/SC) or [CCID](https://en.wikipedia.org/wiki/CCID_(protocol)) interface for applications to hook into.

I'm assuming card means smart card in this case.

**Comment by monich** [comment]
> I'm assuming card means smart card in this case

AFAIU Card Emulation assumes ISO-DEP protocol which is essentially an NFC transport for smart cards. In other words, that would allow to emulate a smart card equipped with an NFC chip

**Comment by monich** [comment]
Ok, now I think it's good to go

**Pull request merged**